### PR TITLE
Added function to remove a component from the system.

### PR DESCRIPTION
**Pull request by daniel-thom**

### CODE REVIEW

File: src/base.jl [hunk 1]
```diff
@@ -280,6 +280,27 @@ function add_component!(sys::System, component::T) where T <: Component
...
 end
 
+"""
+    remove_component!(sys::System, component::T) where T <: Component
+
+Remove a component from the system.
+
+Throws InvalidParameter if the component is not stored.
+"""
+function remove_component!(sys::System, component::T) where T <: Component
```
Comment (jd-lara) [inline src/base.jl]: So to remove the component this will require running a `get_component` first and then select which to remove

Comment (daniel-thom) [reply src/base.jl]: We could add another method that allows you to specify a name.  But if you have the name then you should have the component.  Another option could be to remove all components of type T.

Comment (jd-lara) [reply src/base.jl]: We can also add a remove components that get a device type

Comment (daniel-thom) [reply src/base.jl]: New methods are now present.

**Pull request merged**